### PR TITLE
Give _sync_tables the ability to persist objects that have no sources 

### PR DESCRIPTION
**Pull request by jeremykubica**
This PR addresses issue #75. Creates an optional parameter that allows `_generate_object_table` to keep any object ids are no longer in the `_source` table but were previously in the `_object` table. It populates all counts for these objects with zero.

### CODE REVIEW

File: src/lsstseries/ensemble.py [hunk 1]
```diff
@@ -475,27 +482,52 @@ def _generate_object_table(self):
...
-        # Add total nobs
-        res[self._nobs_col] = counts.groupby(self._id_col).agg("sum")
+        # Add total nobs by summing across each band.
+        res[self._nobs_col] = res.sum(axis=1)
 
         return res
 
-    def _sync_tables(self):
-        """Sync operation to align both tables"""
+    def _sync_tables(self, keep_zero_entries=False):
```
Comment (dougbrn) [inline src/lsstseries/ensemble.py:511]: With this as a kwarg for a hidden function, should we add an ensemble flag to control this? Ensemble.keep_empty_objects or something? If we do, small additional note that the first generation of table from source in read_parquet will always want keep_zero_entries set to False, regardless of the flag, as we wouldn't have an existing ensemble._object to compare to.

Comment (jeremykubica) [reply src/lsstseries/ensemble.py:511]: That's a really good call. I was wondering what the best way to handle turning it off and on (and clearly decided to hold off making that decision). I'm happy to change the code to enable by flag though.

Comment (jeremykubica) [reply src/lsstseries/ensemble.py]: I've updated the code.

File: src/lsstseries/ensemble.py [hunk 2]
```diff
@@ -475,21 +479,46 @@ def _generate_object_table(self):
...
+                # Create a dataframe of the missing IDs with zeros for all bands and counts.
+                rows = {self._id_col: missing_inds}
+                for i in res.columns.values:
+                    rows[i] = [0] * len(missing_inds)
+
+                zero_pdf = pd.DataFrame(rows, dtype=int).set_index(self._id_col)
+                zero_ddf = dd.from_pandas(zero_pdf, sort=True, npartitions=1)
+
+                # Concatonate the zero dataframe onto the results.
+                res = dd.concat([res, zero_ddf], interleave_partitions=True).astype(int)
```
Comment (dougbrn) [inline src/lsstseries/ensemble.py:501]: My only other comment is how this creates a new partition for the zero source objects. I _think_ the preferred option for now would be to append these to the last partition? My thinking being this is the object table, which is generally much smaller than the source table. In many cases I think users will only need a single partition for their object table, so appending them would preserve that. For cases where we need many partitions for the object table I think we would eventually have them being appended by division, where each index is slotted into an existing partition using the divisions as guidelines. But we're not there yet on that end. Just my thinking on this though.

File: src/lsstseries/ensemble.py (hunk 1 above)
Comment (dougbrn) [reply src/lsstseries/ensemble.py]: Nice, looks good!

**Review by dougbrn — APPROVED** [review]
Looks great, thanks Jeremy!

**Pull request merged**